### ISSUE TITLE
Fails to type check annotated code

### ISSUE THREAD

**Issue by Gozala** [jaredly/jerd#7]
Here is an example of the code:

```ts
type Just<T> = {
  x: T
}
enum Maybe<T> {
  None,
  Just<T>
}

const delta = (a:int, b:int):Maybe<int> => {
  if a == b {
    None
  } else {
    Just{x: b - a}
  }
}
```

<img width="528" alt="Screen Shot 2021-03-27 at 12 39 37 AM" src="https://user-images.githubusercontent.com/21236/112713934-10ea6480-8e95-11eb-8ca3-988e8383b3f0.png">

<img width="531" alt="image" src="https://user-images.githubusercontent.com/21236/112713964-3ecfa900-8e95-11eb-9f4e-f39765ce87de.png">

Only thing that seems to works is

<img width="526" alt="image" src="https://user-images.githubusercontent.com/21236/112714011-70e10b00-8e95-11eb-9778-8c8a69881558.png">

**Comment by jaredly** [comment]
That's right, records must be explicitly "upgraded" to specific enum types (because a given record might be part of any number of enums). This fits with my "everything should be explicit in the typed tree" ideology.
I admit this is fairly cumbersome at the moment given that we don't yet have any type inference in the editor.
